uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java

```diff
@@ -321,17 +321,19 @@ public void setMqttTimeoutSec(int mqttTimeoutSec) {
      * Subscribe the MQTT topics by filter.
      *
      * @param clientDeviceId the user defined client device id
-     * @param filter the topics filter to subscribe
+     * @param topicFilterString the topics filter to subscribe
      * @param qos the max value of MQTT QoS for subscribe
      * @throws StatusRuntimeException thrown on gRPC errors
      * @throws IllegalArgumentException on invalid QoS argument
      */
     @When("I subscribe {string} to {string} with qos {int}")
-    public void subscribe(String clientDeviceId, String filter, int qos) {
+    public void subscribe(String clientDeviceId, String topicFilterString, int qos) {
         // getting connectionControl by clientDeviceId
         final String clientDeviceThingName = getClientDeviceThingName(clientDeviceId);
         ConnectionControl connectionControl = getConnectionControl(clientDeviceThingName);
 
+        final String filter = scenarioContext.applyInline(topicFilterString);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;topicFilterString&#39; (index: 1 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;qos&#39;. The same parameter type is validated here for example: aws-greengrass-client-device-auth/uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:251. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).

Reply:
Fixed 0730bc8aca7861b692859a5f3b9ceff75d063a4e